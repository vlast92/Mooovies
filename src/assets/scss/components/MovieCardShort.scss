@import '../variables';

.Movie-card-short {
	text-align: left;
	padding: 1rem;
	border-bottom: 0.063rem solid #d2cbcb;
	background-color: $movie-card-short-background-color;
	color: #000;
	text-decoration: none;

	&:nth-last-of-type(1) {
		border-bottom: none;
	}
	&:hover {
		background-color: darken($movie-card-short-background-color, 20%);
	}
	.movie-poster {
		flex-basis: 2rem;
	}
	.movie-info {
		flex: 1;
		padding: 0 0.75rem;

		.movie-title {
			font-weight: bold;
		}
		.movie-year {
			font-size: 0.8rem;
			color: grey;
		}
	}
}
