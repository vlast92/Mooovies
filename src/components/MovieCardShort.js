import React from "react";
import {Link} from "react-router-dom";
import PropTypes from 'prop-types';

function MovieCardShort(props) {

    const {movie, onClick} = props;

    return (
        <div className="movies-list-item" key={movie.imdbID}>
            <Link to={`/${movie.imdbID}`} onClick={onClick}>
                { movie.Poster !== 'N/A' ?
                    <img src={movie.Poster} style={{maxWidth: '5rem'}} alt={movie.Title}/> : '' }
                {movie.Title} Year: {movie.Year}
            </Link>
        </div>
    );
}

export default MovieCardShort;

MovieCardShort.propTypes = {
    movie: PropTypes.object.isRequired,
    onClick: PropTypes.func
};