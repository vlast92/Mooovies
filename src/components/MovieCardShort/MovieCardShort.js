import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import gridStyles from '../../scss/Grid.module.scss';
import componentStyles from './MovieCardShort.module.scss';

function MovieCardShort(props) {
	const { movie, onClick } = props;

	return (
		<Link
			to={`/movie-details/${movie.imdbID}`}
			onClick={onClick}
			className={`${componentStyles.component} ${gridStyles.flex} ${gridStyles.alignItemsCenter}`}
			key={movie.imdbID}>
			{movie.Poster !== 'N/A' ? (
				<div className={componentStyles.poster}>
					<img src={movie.Poster} alt={movie.Title} />
				</div>
			) : (
				''
			)}
			<div className={componentStyles.info}>
				<div className={componentStyles.title}>{movie.Title}</div>
				<div className={componentStyles.year}>Year: {movie.Year}</div>
			</div>
		</Link>
	);
}

export default MovieCardShort;

MovieCardShort.propTypes = {
	movie: PropTypes.object.isRequired,
	onClick: PropTypes.func,
};
