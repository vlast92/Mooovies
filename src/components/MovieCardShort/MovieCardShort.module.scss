@import './src/scss/variables';

.component {
	text-align: left;
	padding: 1rem;
	border-bottom: 0.063rem solid #d2cbcb;
	background-color: $movie-card-short-background-color;
	color: #000;
	text-decoration: none;

	&:nth-last-of-type(1) {
		border-bottom: none;
	}
	&:hover {
		background-color: darken($movie-card-short-background-color, 20%);
	}
	.poster {
		flex-basis: 2rem;
	}
	.info {
		flex: 1;
		padding: 0 0.75rem;

		.title {
			font-weight: bold;
		}
		.year {
			font-size: 0.8rem;
			color: grey;
		}
	}
}
