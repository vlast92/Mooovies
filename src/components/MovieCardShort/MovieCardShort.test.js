import renderer from 'react-test-renderer';
import React from 'react';

import MovieCardShort from './MovieCardShort';
import { MemoryRouter, Route, Switch } from 'react-router-dom';

test('MovieCardShort renders correctly', () => {
	const dummyMovieInfo = {
		Poster:
			'https://m.media-amazon.com/images/M/MV5BOWVjZTIzNDYtNTBlNC00NTJjLTkzOTEtOTE0MjlhYzI2YTcyXkEyXkFqcGdeQXVyNTAyODkwOQ@@._V1_SX300.jpg',
		Title: 'Blade II',
		Type: 'movie',
		Year: '2002',
		imdbID: 'tt0187738',
	};
	const tree = renderer
		.create(
			<MemoryRouter initialEntries={['/']}>
				<Switch>
					<Route path="/">
						<MovieCardShort movie={dummyMovieInfo} />
					</Route>
				</Switch>
			</MemoryRouter>
		)
		.toJSON();
	expect(tree).toMatchSnapshot();
});
