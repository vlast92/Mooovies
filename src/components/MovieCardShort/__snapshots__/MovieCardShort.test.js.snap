// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCardShort renders correctly 1`] = `
<a
  className="component flex alignItemsCenter"
  href="/movie-details/tt0187738"
  onClick={[Function]}
>
  <div
    className="poster"
  >
    <img
      alt="Blade II"
      src="https://m.media-amazon.com/images/M/MV5BOWVjZTIzNDYtNTBlNC00NTJjLTkzOTEtOTE0MjlhYzI2YTcyXkEyXkFqcGdeQXVyNTAyODkwOQ@@._V1_SX300.jpg"
    />
  </div>
  <div
    className="info"
  >
    <div
      className="title"
    >
      Blade II
    </div>
    <div
      className="year"
    >
      Year: 
      2002
    </div>
  </div>
</a>
`;
